INSTALL.md

```diff
@@ -58,6 +60,11 @@ These include:
 
     CXX                 C++ compiler
     CXXFLAGS            C++ compiler flags
+        * SLATE_HAVE_MT_BCAST uses multiple OMP threads for MPI broadcast communication.
```

Comment:
This looks good, but should probably explain why a user would enable or disable it, i.e., better performance but causes hangs on some systems, particularly Frontier.